### DISCUSSION TITLE
Config to override pages directory for multi tenant applications

### DISCUSSION THREAD

**Discussion by gerhardsletten** [withastro/roadmap#441] [Proposal]
Are looking into ways to use Astro as a `multi tenant` application, where the same source can serve an website that in multiple countries, where each country will have localized url's to different pages.

One way to archive this could be to be able to change the `pages` directory with a config-settings, so each site can live under each area like this:

```
src
-- pages
---- us
------ index.astro
------ product-page.astro
---- de
------ index.astro
------ produktseite.astro
---- es
------ index.astro
------ pagina-del-producto.astro
```

Looks like it can be archived by updating `resolvePages` in [/core/util.ts](https://github.com/withastro/astro/blob/4a1cabfe6b9ef8a6fbbcc0727a0dc6fa300cedaa/packages/astro/src/core/util.ts#L103)

```
function resolvePages(config) {
  return new URL("./pages", config.srcDir);
}
```

**Comment by florian-lefebvre** [comment]
This is now covered by the built-in i18n routing: https://docs.astro.build/en/guides/internationalization/
